cognite/client/_api/geospatial.py

```diff
@@ -0,0 +1,531 @@
+
+        Returns:
+            FeatureTypeList: List of feature types
+
+        Examples:
+
+            Iterate over feature type definitions:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> for feature_type in c.geospatial.list_feature_types():
+                ...     feature_type # do something with the feature type definition
+        """
+        return self._list(method="POST", cls=FeatureTypeList, resource_path=f"{self._RESOURCE_PATH}/featuretypes")
+
+    def retrieve_feature_types(self, external_id: Union[str, List[str]] = None) -> FeatureTypeList:
+        """`Retrieve feature types`
+        <https://docs.cognite.com/api/v1/#operation/getFeatureTypesByIds>
+
+        Args:
+            external_id (Union[str, List[str]]): External ID
+
+        Returns:
+            FeatureTypeList: Requested Type or None if it does not exist.
+
+        Examples:
+
+            Get Type by external id:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.geospatial.retrieve_feature_types(external_id="1")
+        """
+        return self._retrieve_multiple(
+            wrap_ids=True,
+            external_ids=external_id,
+            cls=FeatureTypeList,
+            resource_path=f"{self._RESOURCE_PATH}/featuretypes",
+        )
+
+    def update_feature_types(self, update: Union[FeatureTypeUpdate, List[FeatureTypeUpdate]] = None) -> FeatureTypeList:
+        """`Patch feature types`
+        <https://docs.cognite.com/api/v1/#operation/updateFeatureTypes>
+
+        Args:
+            update (Union[FeatureTypeUpdate, List[FeatureTypePatch]]): the update to apply
+
+        Returns:
+            FeatureTypeList: The updated feature types.
+
+        Examples:
+
+            Add one property to a feature type:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.geospatial.update_feature_types(update=FeatureTypeUpdate(external_id="wells", add=PropertyAndSearchSpec(properties={"altitude": {"type": "DOUBLE"}})))
+        """
+        if isinstance(update, FeatureTypeUpdate):
+            update = [update]
+
+        mapper = lambda it: {
+            "properties": None if not hasattr(it, "properties") else {"add": it.properties},
+            "searchSpec": None if not hasattr(it, "search_spec") else {"add": it.search_spec},
+        }
+        json = {"items": [{"externalId": it.external_id, "update": mapper(it.add)} for it in update]}
+        res = self._post(url_path=f"{self._RESOURCE_PATH}/featuretypes/update", json=json)
+        return FeatureTypeList._load(res.json()["items"], cognite_client=self._cognite_client)
+
+    def create_features(
+        self, feature_type: FeatureType, feature: Union[Feature, List[Feature]], allow_crs_transformation: bool = False
```

Comment:
I think it would be better to only require passing the `feature_type_external_id`, since thats all we need. Also more in line with how things are done elsewhere in the SDK, i.e. we pass around asset_external_ids not `Asset`s.
```suggestion
        self, feature_type_external_id: str, feature: Union[Feature, List[Feature]], allow_crs_transformation: bool = False
```